test/test_SW_VegProd.cc

```diff
@@ -159,4 +159,604 @@ namespace {
+
+        double inputValues[8] = {.0567, .2317, .0392, .0981,
+                                .3218, .0827, .1293, .0405};
+        double shrubLimit = .2;
+
+        // Array holding only grass values
+        double grassOutput[3]; // 3 = Number of grass variables
+
+        // Array holding all values from the estimation
+        double RelAbundanceL0[8]; // 8 = Number of types
+
+        // Array holding all values from estimation minus grasses
+        double RelAbundanceL1[5]; // 5 = Number of types minus grasses
+
+        double SumGrassesFraction = SW_MISSING;
+        double C4Variables[3];
+        double RelAbundanceL0Expected[8] = {0.0567, 0.2317, .0392,
+            .0981, .3218, .0827, .1293, .0405};
+        double RelAbundanceL1Expected[5] = {.1293, .0827, .2884, .4591, .0405};
+
+        Bool fillEmptyWithBareGround = swTRUE;
+        Bool inNorthHem = swTRUE;
+        Bool warnExtrapolation = swTRUE;
+
+        int deallocate = 0;
+        int allocate = 1;
+        int index;
+
+        // RelAbundanceL0 and inputValues indices
+        int succIndex = 0;
+        int forbIndex = 1;
+        int C3Index = 2;
+        int C4Index = 3;
+        int grassAnn = 4;
+        int shrubIndex = 5;
+        int treeIndex = 6;
+        int bareGround = 7;
+
+        // RelAbundanceL1 indices
+        int treeIndexL1 = 0;
+        int forbIndexL1 = 1;
+        int shrubIndexL1 = 2;
+        int grassesIndexL1 = 3;
+        int bareGroundL1 = 4;
+
+        // Reset "SW_Weather.allHist"
+        SW_WTH_read();
+
+        // Allocate arrays needed for `calcSiteClimate()` and `averageClimateAcrossYears()`
+        allocDeallocClimateStructs(allocate, 31, &climateOutput, &climateAverages);
+
+        // Calculate climate of the site and add results to "climateOutput"
+        calcSiteClimate(SW_Weather.allHist, 31, 1980, inNorthHem, &climateOutput);
+
+        // Average values from "climateOutput" and put them in "climateAverages"
+        averageClimateAcrossYears(&climateOutput, 31, &climateAverages);
+
+        // Set C4 results, standard deviations are not needed for estimating vegetation
+        C4Variables[0] = climateAverages.minTemp7thMon_C;
+        C4Variables[1] = climateAverages.ddAbove65F_degday;
+        C4Variables[2] = climateAverages.frostFree_days;
+
+        // Estimate vegetation based off calculated variables and "inputValues"
+        estimatePotNatVegComposition(climateAverages.meanTemp_C, climateAverages.PPT_cm,
+            climateAverages.meanTempMon_C, climateAverages.PPTMon_cm, inputValues, shrubLimit,
+            SumGrassesFraction, C4Variables, fillEmptyWithBareGround, inNorthHem, warnExtrapolation,
+            grassOutput, RelAbundanceL0, RelAbundanceL1);
+
+        // All values in "RelAbundanceL0" should be exactly the same as "inputValues"
+        for(index = 0; index < 8; index++) {
+            EXPECT_NEAR(RelAbundanceL0[index], RelAbundanceL0Expected[index], tol6);

```

Comment:
The comment `// All values in "RelAbundanceL0" should be exactly the same as "inputValues"` spells out the expectation -- please add `EXPECT_NEAR(RelAbundanceL0[index], inputValues[index], tol6);`